factory/shardingFactory.go

```diff
@@ -100,7 +106,13 @@ func CreateNodesCoordinator(
 		return nil, err
 	}
 	if shardIDAsObserver == core.DisabledShardIDAsObserver {
-		shardIDAsObserver = uint32(0)
+		pubKeyBytes, err := pubKey.ToByteArray()
+		if err != nil {
+			log.Warn("cannot get public key bytes", "error", err)
```

Comment:
I think the error escalates anyway. Might wrap the error for more context.

Reply:
done